packages/components/nodes/ChainLink/ChainLink.ts

```diff
@@ -0,0 +1,270 @@
+			...networkProviderCredentials
+		] as INodeParams[];
+        this.inputParameters = [
+			{
+				label: 'Pair',
+				name: 'pair',
+				type: 'asyncOptions',
+				loadMethod: 'getPairAddress',
+                show: {
+                    'actions.operation': ['getPriceFeeds']
+                }
+			},
+            {
+				label: 'Reserve',
+				name: 'reserve',
+				type: 'asyncOptions',
+				loadMethod: 'getReserveAddress',
+                show: {
+                    'actions.operation': ['getProofReserve']
+                }
+			},
+            {
+				label: 'NFT Collection',
+				name: 'nftCollection',
+				type: 'asyncOptions',
+				loadMethod: 'getNftCollection',
+                show: {
+                    'actions.operation': ['getNFTFloorPricing']
+                }
+			},
+		] as INodeParams[];
+	};
+
+	loadMethods = {
+
+        async getNetworkProviders(nodeData: INodeData): Promise<INodeOptionsValue[]> {
+			const returnData: INodeOptionsValue[] = [];
+
+			const networksData = nodeData.networks;
+            if (networksData === undefined) return returnData;
+
+			const network = networksData.network as NETWORK;
+			return getNetworkProvidersList(network);
+		},
+
+        async getPairAddress(nodeData: INodeData): Promise<INodeOptionsValue[]> {
+			const networksData = nodeData.networks;
+            if (networksData === undefined) return [];
+
+			const network = networksData.network as NETWORK;
+
+            return await getInputParametersData(network, "default");
+		},
+
+        async getNftCollection(nodeData: INodeData): Promise<INodeOptionsValue[]> {
+			const networksData = nodeData.networks;
+            if (networksData === undefined) return [];
+            
+			const network = networksData.network as NETWORK;
+
+            return await getInputParametersData(network, "nftFloor");
+		},
+        
+        async getReserveAddress(nodeData: INodeData): Promise<INodeOptionsValue[]> {
+			const networksData = nodeData.networks;
+            if (networksData === undefined) return [];
+            
+			const network = networksData.network as NETWORK;
+
+            return await getInputParametersData(network, "por");
+		},
+	}
+
+	async run(nodeData: INodeData): Promise<INodeExecutionData[] | null> {
+
+		const networksData = nodeData.networks;
+		const actionsData = nodeData.actions;
+        const inputParametersData = nodeData.inputParameters;
+		const credentials = nodeData.credentials;
+
+        if (networksData === undefined || actionsData === undefined || inputParametersData === undefined) {
+            throw new Error('Required data missing');
+        }
+		
+		try {
+			const network = networksData.network as NETWORK;
+
+			const provider = await getNetworkProvider(
+				networksData.networkProvider as NETWORK_PROVIDER,
+				network,
+				credentials,
+				networksData.jsonRPC as string,
+				networksData.websocketRPC as string,
+			)
+
+			if (!provider) throw new Error('Invalid Network Provider');
```

Comment:
though in this case it doesnt matter much but short hand if statement potentially make debugging difficult

```ts
if(condition)   
doSomething() // may run depends on condition
```

one day you want to trouble shoot

```ts
if(condition) 
// add console.log for debugging  
console.log("something") // may run depends on condition
doSomething() // will run regardless of condition`
```

Reply:
yeah I had it this way cus its cleaner. What do you suggest?
```
if(condition) {
  doSomething() // may run depends on condition
}
```

Reply:
This is good, it has scope bracket